tests/conformance/scripts/run-tests/runners/TestRunner.ts

```diff
@@ -20,9 +41,81 @@ export abstract class ReadTestRunner {
    * Execute the reader test. This may involve calling out to separate process, e.g. with
    * `child_process.exec`.
    * @param filePath A path to a `.mcap` file that should be read.
-   * @returns A JSON-encoded object representing the input file.
+   * @returns an IndexedReadTestResult object.
    */
-  abstract runReadTest(filePath: string, variant: TestVariant): Promise<string>;
+  abstract runReadTest(filePath: string): Promise<IndexedReadTestResult>;
+
+  expectedResult(testCase: TestCase): IndexedReadTestResult {
```

Comment:
Seems like this could be `static` or not be a class method at all?

Reply:
static makes sense - i wanted to associate it with the class somehow.